### PR TITLE
integration: always check error for function calls

### LINKED ISSUE #2981: TestV2WatchKeyInDir-2 fail
=== RUN TestV2WatchKeyInDir-2
panic: HTTP body JSON parse error: unexpected end of JSON input: 
goroutine 112949 [running]:
github.com/coreos/etcd/integration.(*testHttpClient).ReadBodyJSON(0xc20a9aeea0, 0xc2154c6120, 0x35)
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/v2_http_kv_test.go:995 +0x2a7
github.com/coreos/etcd/integration.func·003()
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/v2_http_kv_test.go:878 +0x1de
created by github.com/coreos/etcd/integration.TestV2WatchKeyInDir
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/v2_http_kv_test.go:880 +0x978
goroutine 1 [chan receive]:
testing.RunTests(0xa47d48, 0xbd5e60, 0x22, 0x22, 0x1)
	/usr/local/go/src/testing/testing.go:556 +0xad6
testing.(*M).Run(0xc2080565f0, 0xbe09e0)
	/usr/local/go/src/testing/testing.go:485 +0x6c
main.main()
	github.com/coreos/etcd/integration/_test/_testmain.go:118 +0x1d5
goroutine 112953 [runnable]:
sync.(*Mutex).Unlock(0xc21c8cd338)
	/usr/local/go/src/sync/mutex.go:82
net/http.(*conn).hijacked(0xc21c8cd2c0, 0x7f0f04604b00)
	/usr/local/go/src/net/http/server.go:133 +0x76
net/http.(*conn).serve(0xc21c8cd2c0)
	/usr/local/go/src/net/http/server.go:1205 +0xb68
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1751 +0x35e
goroutine 112952 [select]:
net/http.(*persistConn).writeLoop(0xc20d9d3080)
	/usr/local/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0xcbc
goroutine 112951 [runnable]:
net/http.(*persistConn).readLoop(0xc20d9d3080)
	/usr/local/go/src/net/http/transport.go:928 +0x9ce
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0xc9f
goroutine 111724 [runnable]:
sync.(*WaitGroup).Wait(0xc216f57160)
	/usr/local/go/src/sync/waitgroup.go:132 +0x169
net/http/httptest.(*Server).Close(0xc216f57130)
	/usr/local/go/src/net/http/httptest/server.go:168 +0x5f
github.com/coreos/etcd/integration.(*member).Terminate(0xc20da14000, 0xc20e3fd290)
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_test.go:822 +0xb6
github.com/coreos/etcd/integration.(*cluster).Terminate(0xc20e3faf20, 0xc20e3fd290)
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/cluster_test.go:531 +0x73
runtime.Goexit()
	/usr/local/go/src/runtime/panic.go:288 +0xf6
testing.(*common).FailNow(0xc20e3fd290)
	/usr/local/go/src/testing/testing.go:329 +0x3d
testing.(*common).Fatal(0xc20e3fd290, 0xc20e95de50, 0x1, 0x1)
	/usr/local/go/src/testing/testing.go:364 +0x76
github.com/coreos/etcd/integration.TestV2WatchKeyInDir(0xc20e3fd290)
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/integration/v2_http_kv_test.go:888 +0xe60
testing.tRunner(0xc20e3fd290, 0xbd6148)
	/usr/local/go/src/testing/testing.go:447 +0xbf
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xa8b
exit status 2
FAIL	github.com/coreos/etcd/integration	310.688s

### PR DESCRIPTION
**Pull request by yichengq**
fixes #2981

### CODE REVIEW

File: integration/v2_http_kv_test.go [hunk 1]
```diff
@@ -72,7 +72,8 @@ func TestV2Set(t *testing.T) {
 	for i, tt := range tests {
 		resp, err := tc.PutForm(fmt.Sprintf("%s%s", u, tt.relativeURL), tt.value)
 		if err != nil {
-			t.Errorf("#%d: err = %v, want nil", i, err)
+			t.Errorf("#%d: put err = %v, want nil", i, err)
```
Comment (xiang90) [inline integration/v2_http_kv_test.go]: we do not need to change this, right?

Comment (yichengq) [inline integration/v2_http_kv_test.go]: not necessary.

File: integration/v2_http_kv_test.go [hunk 2]
```diff
@@ -182,7 +182,11 @@ func TestV2CreateUpdate(t *testing.T) {
 	}
 
 	for i, tt := range tests {
-		resp, _ := tc.PutForm(fmt.Sprintf("%s%s", u, tt.relativeURL), tt.value)
+		resp, err := tc.PutForm(fmt.Sprintf("%s%s", u, tt.relativeURL), tt.value)
+		if err != nil {
+			t.Errorf("#%d: put err = %v, want nil", i, err)
```
Comment (xiang90) [inline integration/v2_http_kv_test.go]: I guess we can change all errorf to fatalf

Comment (yichengq) [inline integration/v2_http_kv_test.go]: generally if there is etcd failure, it cannot get response in the future too. changed.

File: integration/v2_http_kv_test.go [hunk 3]
```diff
@@ -768,12 +789,18 @@ func TestV2Watch(t *testing.T) {
 	u := cl.URL(0)
 	tc := NewTestClient()
 
-	watchResp, _ := tc.Get(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar?wait=true"))
+	watchResp, err := tc.Get(fmt.Sprintf("%s%s", u, "/v2/keys/foo/bar?wait=true"))
+	if err != nil {
+		t.Fatalf("watch err (%v)", err)
```
Comment (xiang90) [inline integration/v2_http_kv_test.go]: can we always use the format like `put err = %v, want nil` ?

**Comment by xiang90** [comment]
LGTM

File: integration/v2_http_kv_test.go (hunk 3 above)
Comment (yichengq) [inline integration/v2_http_kv_test.go]: no preference. changed.

**Pull request merged**